aws/resource_aws_servicecatalog_portfolio_principal_association.go

```diff
@@ -0,0 +1,165 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"strings"
+	"time"
+)
+
+func resourceAwsServiceCatalogPortfolioPrincipalAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogPortfolioPrincipalAssociationCreate,
+		Read:   resourceAwsServiceCatalogPortfolioPrincipalAssociationRead,
+		Update: resourceAwsServiceCatalogPortfolioPrincipalAssociationUpdate,
+		Delete: resourceAwsServiceCatalogPortfolioPrincipalAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"portfolio_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"principal_arn": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
please add validation here using `ValidateFunc: validateArn,`

Reply:
done (check-in to come)